blog/2020-11-05-HowMuchFasterIsJava15.adoc

```diff
@@ -0,0 +1,105 @@
+
+This article is a follow up on our
+https://www.optaplanner.org/blog/2019/01/17/HowMuchFasterIsJava11.html[ Java 11]
+and our
+https://www.optaplanner.org/blog/2014/03/20/HowMuchFasterIsJava8.html[Java 8] performance comparison benchmarks.
+
+== Benchmark methodology
+
+To run the benchmark we used:
+
+* A stable machine without any other computational demanding processes running and with
+`Intel® Xeon® Silver 4116 @ 2.1 GHz (12 cores total / 24 threads)` and `128 GiB` RAM memory, running `RHEL 8 x86_64`.
+
+* Both G1 and Parallel GC for both Java versions to compare the impact of garbage collection.
+Java executed with the parameters `-Xmx1536M -server -XX:+UseG1GC` and `-Xmx1536M -server -XX:+UseParallelGC` respectively.
+
+* Both OpenJDK 11 version "11.0.6"
+OpenJDK Runtime Environment 18.9 (build 11.0.6+10-LTS)
+OpenJDK 64-Bit Server VM 18.9 (build 11.0.6+10-LTS, mixed mode)
+
+* Both OpenJDK 15 version "15.0.1"
+OpenJDK Runtime Environment 18.9 (build 15.0.1+9-LTS)
+OpenJDK 64-Bit Server VM 18.9 (build 15.0.1+9-LTS, mixed mode)
+
+* OptaPlanner `7.44.0.Final`
+
+* Solving a planning problem involves *no IO* (except a few milliseconds during startup to load the input). *A single
+CPU is completely saturated.* It constantly creates many short-lived objects, and the GC collects them afterwards.
+
+* Each run solves 11 planning problems with OptaPlanner. Each planning problem runs for 5 minutes and starts with a
+30 second JVM warm up which is discarded.
+
+* The benchmarks measure the number of scores calculated per millisecond. Higher is better. Calculating
+a score for a proposed planning solution is non-trivial: it involves many calculations, including checking for
+conflicts between every entity and every other entity.
+
+== Executive summary
+
+With Java 15, the average improvement is *10.5%* for G1 and *6.2%* for Parallel GC. The difference between the two
+garbage collectors running on JDK 15 is *10.7%* leaning in favor of Parallel GC.
+For more information about difference between various GC algorithms, please see
+https://dzone.com/articles/choosing-the-best-garbage-collection-algorithm-for[this article].
+
+The Parallel GC is the superior GC for OptaPlanner than G1, since the throughput is still the most relevant factor when it comes to garbage collection.
+
+== Results
+
+=== Java 11 vs. Java 15
+
+image::JDK11andJDK15UsingG1GC.svg[]
+
+[#table1]
+.Score calculation count per second with G1 GC
+|===
+| 2+^.^|Cloud balancing 2+^.^|Machine reassignment 2+^.^|Course scheduling 2+^.^|Exam scheduling 2+^.^|Nurse rostering^.^|Traveling Tournament
+|JDK ^|200c ^|800c ^|B1 ^|B10 ^|c7 ^|c8 ^|s2 ^|s3 ^|m1 ^|mh1 ^|nl14
+|OpenJDK 11 >|77,073 >|75,343 >|239,262 >|37,991 >|6,861 >|8,577 >|12,748 >|8,375 >|3,047 >|2,872 >|1,905
+|JDK 15 >|79,275 >|76,276 >|298,923 >|39,883 >|7,694 >|10,049 >|14,438 >|9,421 >|3,384 >|3,155 >|2,015
+|Difference >|2.9% >|1.2% >|25.0% >|5.0% >|12.1% >|17.2% >|13.3% >|12.5% >|11.0% >|9.9% >|5.8%
```

Comment:
Just a suggestion can we move % to the Difference label in a column - it is much easier to copy and work with the table when you do not need to cut % of. Also, I suggest to do the same with time units - from a first glance it is hard to distinguish if 77,073 is a seventy-seven or seventy-seven hundreds(this notation could be different in some countries)

Reply:
I slightly prefer the existing number format: 77,073 is more readable on the blog post than 77073. If somebody is copy-pasting the values, spreadsheets can automatically recognize the format.